app/src/main/java/com.example.weatherapp/WeatherAdapter.kt

```diff
@@ -0,0 +1,65 @@
+package com.example.weatherapp
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.bumptech.glide.request.RequestOptions
+import com.example.weatherapp.data.model.ForecastModel
+import com.example.weatherapp.databinding.WeatherCardItemBinding
+import com.example.weatherapp.utils.Constants
+
+class WeatherAdapter(private val days: List<ForecastModel>) :
+    RecyclerView.Adapter<WeatherAdapter.WeatherViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
+        WeatherViewHolder(
+            LayoutInflater.from(parent.context).inflate(R.layout.weather_card_item, parent, false)
+        )
+
+    override fun onBindViewHolder(holder: WeatherViewHolder, position: Int) {
+        holder.bind(days.elementAt(position))
+    }
+
+    override fun getItemCount(): Int = days.size
+
+    class WeatherViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        private val binding = WeatherCardItemBinding.bind(itemView)
+        fun bind(item: ForecastModel) {
+            val requestOptions = RequestOptions().placeholder(R.drawable.default_image)
+            binding.apply {
+                Glide.with(itemView.context)
```

Comment:
Create an extenion for Glide as well, so we encapsulate the implementation of the image renderer

Reply:
Extension added